src/main/java/edu/si/trellis/cassandra/TrellisCassandraApplication.java

```diff
@@ -4,43 +4,163 @@
+    	+ " PRIMARY KEY (identifier, chunk_index)) WITH CLUSTERING ORDER BY (chunk_index ASC);";
+
+    private Session session = null;
+
+    @Override
+    public void initialize(final Bootstrap<TrellisConfiguration> bootstrap) {
+      super.initialize(bootstrap);
+      // Enable variable substitution with environment variables
+      bootstrap.setConfigurationSourceProvider(
+          new SubstitutingSourceProvider(bootstrap.getConfigurationSourceProvider(),
+                                         new EnvironmentVariableSubstitutor(false))
+      );
+    }
+    
+    private Session getCassandraSession() {
+    	if(this.session != null) {
+    		return this.session;
+    	}
+    	Map<String, Object> extraConfig = config.any();
+        Session session = null;
+        boolean newKeyspace = false;
+        try {
+            session = connect(extraConfig, "Trellis");
+        } catch(InvalidQueryException e) {  // No such keyspace
+            createKeyspaceIfNotExists(extraConfig);
+            newKeyspace = true;
+            session = connect(extraConfig, "Trellis");
+        }
+        Metadata metadata = session.getCluster().getMetadata();
+        log.info("Connecting to cluster: {}", metadata.getClusterName());
+        log.info("with nodes: {}", metadata.getAllHosts());
+        if(newKeyspace) {
+        	log.info("Created the keyspace 'Trellis' and associated tables.");
+        } else {
+        	log.info("Using the existing 'Trellis' keyspace.");
+        }
+        this.session = session;
+        return this.session;
+    }
 
     @Override
-    protected <T extends ResourceService> T buildResourceService(IdentifierService idService,
+    protected ResourceService buildResourceService(IdentifierService idService,
                     MementoService mementoService, EventService notificationService) {
-        Map<String, Object> extraConfig = config.any();
-        String cassandraAddress = (String) extraConfig.get("cassandraAddress");
-        Integer cassandraPort = (Integer) extraConfig.get("cassandraPort");
-        Cluster cluster = Cluster.builder().withoutJMXReporting().withoutMetrics().addContactPoint(cassandraAddress)
-                        .withPort(cassandraPort).build();
-        cluster.getConfiguration().getCodecRegistry().register(iriCodec, datasetCodec, bigint(), InstantCodec.instance);
-        Session session = cluster.connect("Trellis");
-        return (T) new CassandraResourceService(session);
+        return new CassandraResourceService(getCassandraSession());
+    }
+
+    @Override
+	protected BinaryService buildBinaryService(IdentifierService idService) {
+    	Map<String, Object> extraConfig = config.any();
+    	Boolean enabled = (Boolean)extraConfig.get("enableCassandraBinaryService");
+		if( enabled.booleanValue() ) {
+			return super.buildBinaryService(idService);
+		} else {
+			return new CassandraBinaryService(idService, getCassandraSession(), 1024*1024 );
```

Comment:
Okay, I may be nuts here, but where is `CassandraBinaryService`? I can't find that class in this PR...